### ISSUE TITLE
Completions not erasing with `complete -c <command> -e`

### ISSUE THREAD

**Issue by donkopotamus** [fish-shell/fish-shell#6716]
I have been investigating a large slowdown in `git` completions in `3.1` on Windows (running under `msys2`), where completion times have ballooned out to between 3-10 seconds for some commands.

In doing so I have noticed that I seem unable to **erase** completion rules in a shell until they have actually been **used**.  For example, in a fresh shell
```
$ sh -c 'env HOME=$(mktemp -d) fish'

Welcome to fish, the friendly interactive shell
Type `help` for instructions on how to use fish
$ fish --version
fish, version 3.1.0

# we think we are erasing the completion rules
$ complete -c git -e

# but this generates a tonne of completions (and is *very* slow)
$ complete -C"git checkout " | wc -l
1524

# try erasing again
$ complete -c git -e

# now it behaves as expected
$ complete -C"git checkout " | wc -l
56
```
I've observed the same behaviour, in terms of rules not being erased until having somehow been used, in `3.0.2` on `RHEL7.5`.  This does not feel like "expected" behaviour.

**Comment by faho** [comment]
Just like functions, completions are autoloaded, so this is to be expected. Before you try them they don't exist.

**Comment by donkopotamus** [comment]
Hmm, I understand.  However is that "expected" from a user point of view?  I may just be extrapolating from my own experience, but If a user puts `complete -c git -e` in their `.config/fish/config.fish`, they are probably expecting that to disable the completion ... as opposed to needing to artificially trigger the completion before disabling it.

(Whether that's possible to implement I don't know)

**Comment by zanchey** [comment]
We do this for functions by adding a tombstone (#213), which seems like a reasonable thing to support to me.

**Comment by rsteube** [comment]
@donkopotamus  seems adding an empty user completion prevents the system one to be autoloaded:
```sh
touch ~/.config/fish/completions/git.fish
```

**Comment by ccoVeille** [comment]
Hi everyone, I'm reactivating this old issue

I found it by trying to understand why the project [carapace](https://github.com/carapace-sh/carapace-bin) was suggesting a strange hack

I opened 
- https://github.com/carapace-sh/carapace-bin/issues/2518

I had a discussion with the maintainer of this project, and we agree the problem cannot be handled at carapace level.

I would like you guys to read the issue I quote, and if possible give us feedback here.

To sum up, the point I raised, and give you a will to read it, here is a TL; DR;

is there any hope to work on this ?

Thanks

**Comment by zanchey** [comment]
Nobody is tackling it as far as I know.

**Comment by ccoVeille** [comment]
Thanks @korbelus

**Comment by ccoVeille** [comment]
Thanks @krobelus
